indexer/processor/encoding.go

```diff
@@ -0,0 +1,24 @@
+package processor
+
+import (
+	"encoding/binary"
+	"math/big"
+)
+
+// DecodeVersionNonce is an re-implementation of Encoding.sol#decodeVersionedNonce.
+// If the nonce is greater than 32 bytes (solidity uint256), bytes [32:] are ignored
```

Comment:
Not sure how hard the test is to write and don't feel super strongly but a test that passes inputs into both this and the solidity version and asserts the same output would be nice

Reply:
yea unfortunately, the bindings dont expose an eth_call to the Encoding.sol smart contract. I made sure the E2E test decodes the nonce and checks it with I think is good enough